dataproc/dataproc.go

```diff
@@ -0,0 +1,313 @@
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Wait for the job to complete
+	fmt.Println("Waiting for job to complete")
+	_, err = waitForJob(service, jobId, *projectID, *region)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Job is finished")
+
+	// Get the job output
+	clusterData, err := getClusterDataByName(service, *clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	output, err := getJobOutput(storageService, *projectID, clusterData[1], clusterData[3], jobId)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("Job output:\n%s\n", string(output))
+
+	// Delete the cluster
+	_, err = deleteCluster(service, *projectID, *region, *clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Cluster deleted")
+	fmt.Printf("Done...")
+}
+
+// getDataprocService creates and returns a Cloud Dataproc service object.
+func getDataprocService() (service *dataproc.Service, err error) {
+	ctx := context.Background()
+	client, err := google.DefaultClient(ctx, scope)
+
+	// Create a new Dataproc service
+	service, err = dataproc.New(client)
+
+	return service, err
+}
+
+// getStorageService creates and returns a Cloud Storage service object.
+func getStorageService() (service *storage.Client, err error) {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+
+	return client, err
+}
+
+// createCluster creates a Cloud Dataproc cluster with the given name and region.
+func createCluster(service *dataproc.Service, name string, region string,
+	project string, zone string) (response string, err error) {
+	// Create a gceConfig object for the cluster
+	gceConfig := dataproc.GceClusterConfig{
+		ZoneUri: fmt.Sprintf(computeUriBase, project, zone),
+	}
+
+	// Create a cluserConfig for the cluster
+	clusterConfig := dataproc.ClusterConfig{
+		GceClusterConfig: &gceConfig,
+	}
+
+	// Create a cluster object
+	cluster := dataproc.Cluster{
+		ClusterName: name,
+		ProjectId:   project,
+		Config:      &clusterConfig,
+	}
+
+	// Create the cluster
+	res, err := service.Projects.Regions.Clusters.Create(project, region, &cluster).Do()
+	return fmt.Sprintf("%s", res), err
+}
+
+// deleteCluster deletes the Cloud Dataproc cluster with the given project, region, and name.
+func deleteCluster(service *dataproc.Service, project string, region string, name string) (response string, err error) {
+	// Delete the cluster
+	res, err := service.Projects.Regions.Clusters.Delete(project, region, name).Do()
+
+	return fmt.Sprintf("%s", res), err
+}
+
+// getClusterDataByName returns metadata about the cluster with the given name.
+func getClusterDataByName(service *dataproc.Service, name string) (clusterData []string, err error) {
+	// Get a list of clusters
+	clusters, err := listClusters(service)
+	if err != nil {
+		return nil, err
+	}
+
+	// Find the cluster requested in the list
+	for _, item := range clusters {
+		if item[0] == name {
+			return item, err
+		}
+	}
+
+	return nil, errors.New("Cluster not found")
```

Comment:
Error strings should start with a lowercase letter. They are often concatenated with a prefix - having a capital in the middle is ugly.